book/controller.rst

```diff
@@ -604,12 +604,13 @@ session.
 Flash Messages
 ~~~~~~~~~~~~~~
 
-You can also store small messages that will be stored on the user's session.
-This is useful when processing a form:
-you want to redirect and have a special message shown on the *next* page.
-These types of messages are called "flash" messages.
+You can also store special messages, called "flash" messages, on the user's
+session. By design, flash messages are meant to be processed exactly once. This
+means that they vanish from the session automatically as soon as they are
+retrieved. This feature makes "flash" messages particularly suited for storing
+user notifications.
 
-For example, imagine you're processing a form submit::
+Consider the following form processing example::
```

Comment:
My question: is `form submit` correct? `submit` is a verb, not a noun. Should it be `form submission`?